urdf/CMakeLists.txt

```diff
@@ -65,13 +86,51 @@ if(BUILD_TESTING)
   ament_cpplint()
   ament_lint_cmake()
   ament_uncrustify(LANGUAGE "C++")
+
+  # Write files to enable pluginlib to find urdf_xml_parser in build folder
+  set(fake_urdf_prefix "${CMAKE_CURRENT_BINARY_DIR}/fake_urdf_install")
+  # Copy urdf_xml_parser to fake prefix lib/ and bin/
+  file(MAKE_DIRECTORY "${fake_urdf_prefix}/lib")
+  file(MAKE_DIRECTORY "${fake_urdf_prefix}/bin")
+  add_custom_command(TARGET urdf_xml_parser
+    POST_BUILD
+    COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_FILE:urdf_xml_parser>
+    "${fake_urdf_prefix}/lib/"
+  )
+  add_custom_command(TARGET urdf_xml_parser
+    POST_BUILD
+    COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_FILE:urdf_xml_parser>
+    "${fake_urdf_prefix}/bin/"
+  )
+  # Fake install plugin description file
+  file(COPY "urdf_parser_description.xml" DESTINATION "${fake_urdf_prefix}/share/urdf/")
+  # Fake install package.xml
+  configure_file("package.xml" "${fake_urdf_prefix}/share/urdf/package.xml" COPYONLY)
+  # Fake install entry in ament_index telling pluginlib where to find plugin description
+  file(WRITE "${fake_urdf_prefix}/share/ament_index/resource_index/urdf_parser_plugin__pluginlib__plugin/urdf"
+    "share/urdf/urdf_parser_description.xml\n")
+  # Fake install entry in ament_index declaring this a package
+  file(WRITE "${fake_urdf_prefix}/share/ament_index/resource_index/packages/urdf" "")
```

Comment:
Well, yuck.  If I'm understanding this properly, this is more-or-less recreating the install directory structure under build.  I guess its OK, but it seems pretty fragile to me; if we change the ament structure for any reason, this is going to break.

How valuable do you think this performance test is?  I'm wondering if it would be more straightforward to not include it at all.

Reply:
> Well, yuck. If I'm understanding this properly, this is more-or-less recreating the install directory structure under build. I guess its OK, but it seems pretty fragile to me; if we change the ament structure for any reason, this is going to break.

Yeah. It will break if how pluginlib uses the ament index is changed. I copied this logic from pluginlib's own tests. A cmake function provided by pluginlib that implemented this would eliminate the duplication if something does change

https://github.com/ros/pluginlib/blob/7c3e7d5b7af6d776079e2e9cba2ac9a5581777b0/pluginlib/CMakeLists.txt#L86-L94

> How valuable do you think this performance test is? I'm wondering if it would be more straightforward to not include it at all.

It's straight forward to delete it, though it does give valuable info about the overhead of the plugin infrastructure, and it does prove the plugin infrastructure works. I would rather robustify it than delete it.

Reply:
That's a valid point @clalancette, these performance tests we just a guide to make sure that the new plugin-based file type check weren't introducing large amounts of overhead. While no one should be parsing URDF/SDF inside their control loop, we don't want this plugin structure to have a substantial impact on existing code execution times.

I think it would be fine to refrain from including this install deployment. This code was more-or-less intended as a one-off verification that the plugin checking times were reasonable. @sloretz can you recap the results of the tests here? Would they be impacted at all by the size of the URDF/SDFormat file being parsed?

Reply:
> I think it would be fine to refrain from including this install deployment. This code was more-or-less intended as a one-off verification that the plugin checking times were reasonable.

I would recommend keeping it as it's the only test that makes sure the urdf plugin is loadable.

> @sloretz can you recap the results of the tests here?

[Linux amd64](https://ci.ros2.org/job/ci_linux/12021/console#console-section-14)
* direct - 0.015 ms
* plugin - 0.743 ms

[Linux aarch64](https://ci.ros2.org/job/ci_linux-aarch64/7045/console#console-section-12)
* direct - 0.045 ms
* plugin - 3.009 ms

[OSX](https://ci.ros2.org/job/ci_osx/9796/console#console-section-2)
* direct - 0.012 ms
* plugin - 2.371 ms

[Windows](https://ci.ros2.org/job/ci_windows/11954/console#console-section-34)
* direct - 0.017 ms
* plugin - 1.970 ms

> Would they be impacted at all by the size of the URDF/SDFormat file being parsed?

Yes since cb348db asks tinyxml2 to parse the whole document to decide the confidence. See discussion on https://github.com/ros2/urdf/pull/13#discussion_r465961287 . The size dependent overhead would apply times every plugin with a similar score plugin - https://github.com/ros/sdformat_urdf/pull/1#discussion_r467937257

Reply:
3341988 uses a new pluginlib CMake function that enables testing plugins in the same package that built them: https://github.com/ros/pluginlib/pull/201